### DISCUSSION TITLE
[work-around] LSPosed not opening (on 2nd launch)/ Shell keeps stopping

### DISCUSSION THREAD

**Discussion by LeeBinder** [LSPosed/LSPosed#2602] [General]
Many people incl. me have been experiencing this issue with many different ROMs, both stock and custom and across many different Android versions, so far w/o any fix ¹, esp. because it appears to be caused by the "parasitic manager" hooking into the shell process, which - as per explanation by a dev in another issue here - will remain as it is.

I'm sharing the fix which worked for me on my secondary phone, a Samsung Galaxy A8 (2018) with stock Android 9 ROM:

1. in Magisk, remove the LSPosed Zygisk module and disable Zygisk in settings
2. reboot
3. in the Magisk modules section, add [Riru](https://github.com/RikkaApps/Riru/releases) and LSPosed Riru
4. reboot
5. open LSPosed Manager via the status bar to test if it launches
6. install a module apk
7. within LSPosed Manager, enable it and activate it for the desired app(s)
8. reboot
9. again open LSPosed Manager. Hopefully it'll open now w/o any shell error, and you will see the activated module
10. if you want to add more modules, do so _one-by-one_, reboot, test

Optional: create a shortcut to your home screen and test, if LSPosed launches from it, too, as supposed.

Possible explanation: it seems the LSPosed Riru flavor parasitic manager uses a more shell-friendly method.

@yujincheng08 maybe you can make this post sticky (at least issues can be pinned on-top) so people can find it and test the work-around, rather than keep re-opening yet another topic on this issue, as has been happening in the past, unfortunately.

Also thanks for continuing to issue Riru based releases, which once more has proven to be very helpful.

--------------
¹
https://github.com/LSPosed/LSPosed/issues/1964
https://github.com/LSPosed/LSPosed/issues/2384
https://github.com/LSPosed/LSPosed/issues/1967
https://github.com/LSPosed/LSPosed/issues/1919
https://github.com/LSPosed/LSPosed/issues/2074

**Comment by LeeBinder** [comment]
@Howard20181 sorry Howard, why on the planet do you close this with the remark "outdated"?? This issue occurs at least up to Android 12, see https://github.com/LSPosed/LSPosed/issues/1964, https://github.com/LSPosed/LSPosed/issues/1967, https://github.com/LSPosed/LSPosed/issues/1919 and https://github.com/LSPosed/LSPosed/issues/2074, and evidently it occurs even with the latest LSPosed Zygisk + latest Magisk....

We all make mistakes, and unfortunately you didn't bother to follow any of the issue links I posted .. :(

Rather than smothering this discussion before it has even begun for people to read it, I am therefore requesting you to undo your mistake and reopen this topic, which is not outdated but current.

Thank you, Howard.

**Reply by Howard20181** [reply]
1. Unconditionally directing to a wrong solution when the root cause of the problem is not clear will prevent the problem from being located and solved
2. Riru has been deprecated, and directing users to migrate back to Riru is not the right thing to do

**Comment by LeeBinder** [comment]
@Howard20181 I am proving why again you are totally wrong with all your arguments in both instances:

> 1. Unconditionally directing to a wrong solution when the root cause of the problem is not clear will prevent the problem from being located and solved

- "_solution_": like my topic title clearly states, I am not offering a "solution" but a work-around. Such cannot be "_wrong_" per-se, and certainly isn't
- "_the root cause of the problem is not clear_": 1. it doesn't matter for this topic. 2. I don't claim to be tackling "_the root cause_". 3. as stated before, the dev team has decided to stick to the parasitic manager hooking into shell for now because other options proved to be even more disadvantageous.
- that me offering an alternative "_will prevent the problem from being located and solved_" is a completely garbled up "understanding" of both context and the rule of cause and effect.
- "_Unconditionally directing_": in general, offering an alternative is neither "_unconditional_", nor is it in any way a "_directive_"....

> 2. Riru has been deprecated, and directing users to migrate back to Riru is not the right thing to do

Howard, obviously you haven't even bothered to look at [LSPosed/releases](https://github.com/LSPosed/LSPosed/releases) in a long time, because then you would've seen that the LSPosed dev team is still issuing Riru based releases - **with each single release**!
Why? Because running Zygote in Magisk simply doesn't always work. Therefore the dev team here has decided to continue maintaining Riru based LSPosed. So much about "_has been deprecated_"...

In that light it's not up to you to decide for users what for them "is not the right thing to do", and what isn't. Not sure where you live, but I live in a liberal democratic world of choices, both geographically and as state of mind.

Howard, I'll be holding up the mirror now so you see how you're acting. you're misusing your privs as a moderator/ maintainer here, arrogantly censoring like a little dictator treating me like an a****le, totally overstretching your competences, using "reasoning" which can hardly be more skewed.

If you have a bad (in this case extremely bad) day, please take some time off and don't bash on other innocent users who are actively, honestly contributing by sharing what they found out. That's absolutely unacceptable.

You're now expected to re-open this "discussion" immediately upon reading this, so it can actually start. If you don't comply with that, I'll file an official complaint against you with the other maintainers/ moderators here after 12 hours of me posting this. Hope that was clear.

(don't even think about continuing to wrongfully censor by removing this post or even the entire discussion - I've saved everything in different formats for documentation)

I'll grant you ONE chance, whereafter I'll forget about how you acted up here today. Be smart and don't miss it...
